### PR TITLE
KUBOS-406 Prevent setting "high level" targets

### PR DESCRIPTION
**Pull request by kyleparrott**
This also filters the available targets to a project based on its platform type. This pr adds a "platform" attribute to a project's module.json. The key has either an rt or linux value. Depending on the platform type it will only allow the corresponding target types to be set.

This includes a 1 line fix for a bug hanging around from #21 that was causing the integration test to fail.

### CODE REVIEW

File: kubos/test/test_sdk_utils.py [hunk 1]
```diff
@@ -103,6 +108,35 @@ def test_link_entities_discovery(self):
...
 
+    def test_get_all_eligible_targets(self):
+        '''
+        setup target hierarchy:
+
+        target_a
+        |_target_b
+          |_target_c
+
+        target_c should be the only target that is an elibible target
```
Comment (CatFreed) [inline kubos/test/test_sdk_utils.py]: nit: spelling

File: kubos/target.py [hunk 2]
```diff
@@ -26,9 +26,12 @@
 from kubos.utils.sdk_utils import *
 
 def addOptions(parser):
-    parser.add_argument('set_target', nargs='?', default=None, help='set a new target board or display the current target')
+    proj_type = get_project_type()
+    choices = load_target_list(proj_type)
+    parser.add_argument('set_target', nargs='?', default=None, choices=choices, help='set a new target board or display the current target')
```
Comment (CatFreed) [inline kubos/target.py]: nit: capitalize "Set"

File: kubos/utils/sdk_utils.py [hunk 3]
```diff
@@ -120,3 +121,109 @@ def link_global_cache_to_project(project):
...
+    '''
+    Returns the project "platform" type: either None, 'linux', or 'rt'
+    '''
+    platform_key = 'platform'
+    valid_platforms = ['rt', 'linux']
+    module_json = os.path.join(os.getcwd(), 'module.json')
+    if os.path.isfile(module_json):
+        with open(module_json, 'r') as module_file:
+            data = json.loads(module_file.read())
+        if platform_key in data:
```
Comment (CatFreed) [inline kubos/utils/sdk_utils.py]: Missing case? if platform_key isn't present, still return None

File: kubos/utils/sdk_utils.py [hunk 4]
```diff
@@ -120,3 +121,114 @@ def link_global_cache_to_project(project):
...
+def get_target_lists():
+    '''
+    Splits rt and linux targets.
+    Returns the list of kubos_rt targets and the list of linux targets.
+    '''
+    linux_list = []
+    rt_list = []
+    target_list = get_all_eligible_targets(GLOBAL_TARGET_PATH)
+
+    #TODO: Get a better way of determining linux targets
```
Comment (CatFreed) [inline kubos/utils/sdk_utils.py]: Per the new standards doc :smile: Is there a JIRA story for this TODO item?

Comment (kyleparrott) [inline kubos/utils/sdk_utils.py]: There is now 😄 [KUBOS-495](https://kubostech.atlassian.net/browse/KUBOS-495)

**Pull request merged**